core/chainlink-sgx.Dockerfile

```diff
@@ -1,19 +1,39 @@
 # Build Chainlink with SGX
-FROM smartcontract/builder:1.0.20 as builder
+FROM smartcontract/builder:1.0.25 as builder
 
 # Have to reintroduce ENV vars from builder image
 ENV PATH /root/.cargo/bin:/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/sgxsdk/bin:/opt/sgxsdk/bin/x64
 ENV LD_LIBRARY_PATH /opt/sgxsdk/sdk_libs
 ENV SGX_SDK /opt/sgxsdk
 
+WORKDIR /chainlink
+COPY GNUmakefile VERSION ./
+COPY tools/bin/ldflags ./tools/bin/
+
+# And yarn likewise
+COPY yarn.lock package.json ./
+COPY explorer/client/package.json ./explorer/client/
+COPY explorer/package.json ./explorer/
+COPY operator_ui/package.json ./operator_ui/
+COPY styleguide/package.json ./styleguide/
+COPY tools/json-api-client/package.json ./tools/json-api-client/
+COPY tools/local-storage/package.json ./tools/local-storage/
+COPY tools/redux/package.json ./tools/redux/
+RUN make yarndep
+
+# Do go mod download in a cacheable step
+ADD go.mod go.sum ./
+RUN go mod download
+
+# Env vars needed for chainlink sgx build
 ARG COMMIT_SHA
 ARG ENVIRONMENT
 ENV SGX_ENABLED yes
 ARG SGX_SIMULATION
 
-WORKDIR /go/src/github.com/smartcontractkit/chainlink
+# Install chainlink
 ADD . ./
```

Comment:
I wouldn't worry about this for this PR, but i feel like we should avoid adding the entire repo as a step

Reply:
Good idea, might put that in another PR.